src/map/pc.cpp

```diff
@@ -6066,22 +6066,18 @@ enum e_setpos pc_setpos(struct map_session_data* sd, unsigned short mapindex, in
 		if (sc && sc->count) { // Cancel some map related stuff.
 			if (sc->data[SC_JAILED])
 				return SETPOS_MAPINDEX; //You may not get out!
-			status_change_end(&sd->bl, SC_BOSSMAPINFO, INVALID_TIMER);
-			status_change_end(&sd->bl, SC_WARM, INVALID_TIMER);
-			status_change_end(&sd->bl, SC_SUN_COMFORT, INVALID_TIMER);
-			status_change_end(&sd->bl, SC_MOON_COMFORT, INVALID_TIMER);
-			status_change_end(&sd->bl, SC_STAR_COMFORT, INVALID_TIMER);
-			status_change_end(&sd->bl, SC_MIRACLE, INVALID_TIMER);
-			if (sc->data[SC_KNOWLEDGE]) {
-				struct status_change_entry *sce = sc->data[SC_KNOWLEDGE];
+			for (const auto &it : status_db) {
+				sc_type status = static_cast<sc_type>(it.first);
+
+				if (sd->sc.data[status] && it.second->flag[SCF_REM_ON_MAPWARP])
+					status_change_end(&sd->bl, status, INVALID_TIMER);
+			}
```

Comment:
Would be nice if we could move the code into it's respective file, while we are it.
Also since there are a few "stop all" status changes kind of functions right now it might be good to create a single function to create a function that takes a "bl" and the respective SCF_REM flag as parameter and handle this in there.